### ISSUE TITLE
Total bundle size calculation is incorrect

### ISSUE THREAD

**Issue by epicfaace** [codalab/codalab-worksheets#3446]
During the check for user disk quota when running `cl upload`, the total bundle size calculation is incorrect, because it adds up the size of all the sources (when, in reality, sometimes the sources may be unpacked; or if there are multiple sources, they'll all be tarred together).

https://github.com/codalab/codalab-worksheets/blob/8ef41cbdce33edd8fae0e6744a2f2043a9c69bb6/codalab/lib/bundle_cli.py#L1370-L1377

We actually can't determine the true file size, because the file is sometimes streamed as it is being created (and we don't want to wait to create the entire file before sending it over). Sometimes, `packed['filesize']` is set, which could be used to solve this issue, but it won't be set once we fully implement https://github.com/codalab/codalab-worksheets/issues/3437.

**Possible solution:** Perhaps, when the file is streaming to the server, once `disk_left` number of bytes have been streamed, we should just stop the upload then.

cc @adiprerepa

**Comment by adiprerepa** [comment]
> **Possible solution:** Perhaps, when the file is streaming to the server, once `disk_left` number of bytes have been streamed, we should just stop the upload then.

Yeah, you might be right. Would we have to delete this partially uploaded bundle off the server too? Or is that handled?

**Comment by percyliang** [comment]
I thought the point was to do a quick check locally before starting the upload, so that the user doesn't have to wait a long time before failure.  Can we not just improve the calculation of the data size (using the same logic as on the server)?

**Comment by epicfaace** [comment]
> Can we not just improve the calculation of the data size (using the same logic as on the server)?

In order to calculate it properly on the client, we'd have to first completely tar and gzip the input files. We won't be able to stream the files if that is the case.

For clarity, if you run:

```
cl upload a.txt b.txt c.txt
```

The client will first compress a.txt, b.txt, and c.txt into a single archive, `contents.tar.gz`, and then stream it to the server.
